vpr/src/route/partition_tree.cpp

```diff
@@ -29,9 +39,12 @@ std::unique_ptr<PartitionTreeNode> PartitionTree::build_helper(const Netlist<>&
     /* Cutlines are placed between integral coordinates.
      * For instance, x_total_before[0] assumes a cutline at x=0.5, so fanouts at x=0 are included but not
      * x=1. It's similar for x_total_after[0], which excludes fanouts at x=0 and includes x=1.
-     * Note that we have W-1 possible cutlines for a W-wide box. */
-    std::vector<int> x_total_before(W - 1, 0), x_total_after(W - 1, 0);
-    std::vector<int> y_total_before(H - 1, 0), y_total_after(H - 1, 0);
+     * Note that we have W-1 possible cutlines for a W-wide box.
+     *
+     * Here, *_total_before holds total score of nets before the cutline and not intersecting it.
+     * In ParaDRo this would be total_before + total_on. (same for total_after)*/
+    std::vector<uint64_t> x_total_before(W - 1, 0), x_total_after(W - 1, 0), x_total_on(W - 1, 0);
```

Comment:
Any reason to use 64-bit ints?  A regular signed 32-bit int will have more than enough dynamic range for nets cut, so I'd switch to this if there's no reason for 64 bits.

Reply:
Artifact from experiments where I was using heap pushes etc. here